File: Server/DataBase/Entities/RecipeIngredientEntity.cs
namespace Server.DataBase.Entities;

public class RecipeIngredientEntity
{
    public int Id { get; init; }
    public int RecipeEntityId { get; init; }
    public string Name { get; set; } = null!;
    public int Quantity { get; set; }
    public string Unit { get; set; } = null!;
    public RecipeEntity RecipeEntity { get; set; }
}

File: client/src/models/RecipeIngredientEntity.js
class RecipeIngredientEntity {
    constructor(json) {
        if (json) {
            this.name = json.name;
            this.quantity = json.quantity;
            this.unit = json.unit;
        } else {
            this.name = '';
            this.quantity = 0;
            this.unit = '';
        }
    }
}

export default RecipeIngredientEntity;